manifests/install.pp

```diff
@@ -65,6 +65,14 @@
 
   case $python::provider {
     pip: {
+      # Install pip without pip, see https://pip.pypa.io/en/stable/installing/.
+      exec { 'bootstrap pip':
+        command => 'curl https://bootstrap.pypa.io/get-pip.py | python',
```

Comment:
This is only done if you choose `pip` as a provider.

Generally I'd agree with you, but I couldn't think of a better way to do this whilst remaining cross-platform compatible.